src/Entity/Query/Sparql/SparqlArg.php

```diff
@@ -138,4 +138,26 @@ public static function serialize($value, $format, $lang = NULL) {
     return $serializer->serialiseValue($data);
   }
 
+  /**
+   * Prefixes a keyword with a prefix in order to be treated as a variable.
+   *
+   * @param string $key
+   *   The name of the variable.
+   * @param bool $blank
+   *   Whether or not to be a blank note.
+   *
+   * @return string
+   *   The variable.
+   */
+  public static function toVar($key, $blank = FALSE) {
+    // Deal with field.property as dots are not allowed in var names.
+    $key = str_replace('.', '_', $key);
```

Comment:
So, as an edge case, `field_a_b.c` can be to mixed up with `field_a.b_c`? Probably is not fixing 100% but what about replacing dot with two underscores?

Want a more safe approach? Drupal fields only accept letters, digits & underscores. But, according to https://www.w3.org/TR/sparql11-query/#rVARNAME, SPARQL variables are also accepting some unicode chars and the middot char '·'. I've tested the middot and works nice in the SPARQL endpoint :)